detection_rules/version_lock.py

```diff
@@ -224,65 +219,73 @@ def add_changes(r, *msg):
 
                 elif min_stack > latest_locked_stack_version:
                     route = 'B'
                     # 3) on the latest stack, locking in a breaking change
                     previous_lock_info = {
-                        "rule_name": existing_rule_lock["rule_name"],
-                        "sha256": existing_rule_lock["sha256"],
-                        "version": existing_rule_lock["version"],
-                        "type": existing_rule_lock["type"]
+                        "rule_name": lock_from_file["rule_name"],
+                        "sha256": lock_from_file["sha256"],
+                        "version": lock_from_file["version"],
+                        "type": lock_from_file["type"]
                     }
-                    existing_rule_lock.setdefault("previous", {})
+                    lock_from_file.setdefault("previous", {})
 
                     # move the current locked info into the previous section
-                    existing_rule_lock["previous"][str(latest_locked_stack_version)] = previous_lock_info
+                    lock_from_file["previous"][str(latest_locked_stack_version)] = previous_lock_info
 
                     # overwrite the "latest" part of the lock at the top level
                     # TODO: would need to preserve space here as well if supporting forked version spacing
-                    existing_rule_lock.update(current_rule_lock, min_stack_version=str(min_stack))
-                    add_changes(
-                        rule,
+                    lock_from_file.update(lock_from_rule, min_stack_version=str(min_stack))
+                    new_version = lock_from_rule['version']
+                    log_changes(
+                        rule, route, new_version,
                         f'previous {latest_locked_stack_version} saved as version: {previous_lock_info["version"]}',
                         f'current min_stack updated to {min_stack}'
                     )
 
                 elif min_stack < latest_locked_stack_version:
                     route = 'C'
                     # 4) on an old stack, after a breaking change has been made (updated fork)
-                    assert str(min_stack) in existing_rule_lock.get("previous", {}), \
+                    assert str(min_stack) in lock_from_file.get("previous", {}), \
                         f"Expected {rule.id} @ v{min_stack} in the rule lock"
 
                     # TODO: Figure out whether we support locking old versions and if we want to
                     #       "leave room" by skipping versions when breaking changes are made.
                     #       We can still inspect the version lock manually after locks are made,
                     #       since it's a good summary of everything that happens
-                    existing_rule_lock["previous"][str(min_stack)] = current_rule_lock
-                    existing_rule_lock.update(current_rule_lock)
```

Comment:
This change was added in #1870 as a fix to #1949, but was actually incomplete (and inaccurate it is resolution). By correctly fixing this bug, this change needed to be reverted (removed).